### PR TITLE
Random stab at fixing test failure

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
SUMMARY: None

#### Purpose of change
Looks like I broke something about lightmap initialization in #30940 
There's a vision test failing consistently now, and it bisects to this change at least.

#### Describe the solution
If this works it's because I accidentally cleared the seen cache at the player's location and this restores it to visibility.

### CODE REVIEW

**Comment by kevingranade** [comment]
That did it, for posterity, what was happening is:
I accidentally cleared the tile the player is standing on in the seen cache.
Shadowcasting doesn't set this tile, so it remained unset, but it doesn't usually cause a problem since the player's avatar is drawn regardless of visibility.
map::apparent_light_helper() checks the tile adjacent to a wall in the direction of the player for visibility to determine if illumination of that side of the wall should contribute to apparent lighting.
If the player is standing adjacent to a wall, that adjacent tile is now "not visible to the player", since shadowcasting never set it. therefore the wall itself does not count as illuminated.

tl;dr hooray for unit tests.
